src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/DoNotGuardDictionaryRemoveByContainsKeyTests.cs

```diff
@@ -0,0 +1,270 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Xunit;
+
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.DoNotGuardDictionaryRemoveByContainsKey,
+    Microsoft.NetCore.Analyzers.Performance.DoNotGuardDictionaryRemoveByContainsKeyFixer>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.DoNotGuardDictionaryRemoveByContainsKey,
+    Microsoft.NetCore.Analyzers.Performance.DoNotGuardDictionaryRemoveByContainsKeyFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Performance.UnitTests
+{
+    public class DoNotGuardDictionaryRemoveByContainsKeyTests
+    {
+        #region Reports Diagnostic
+        [Fact]
+        public async Task RemoveIsTheOnlyStatement_ReportsDiagnostic_CS()
+        {
+            string source = @"
+" + CSUsings + @"
+namespace Testopolis
+{
+    public class MyClass
+    {
+        private readonly Dictionary<string, string> MyDictionary = new Dictionary<string, string>();
+
+        public MyClass()
+        {
+            if ([|MyDictionary.ContainsKey(""Key"")|])
+                MyDictionary.Remove(""Key"");
+        }
+    }
+}";
+
+            string fixedSource = @"
+" + CSUsings + @"
+namespace Testopolis
+{
+    public class MyClass
+    {
+        private readonly Dictionary<string, string> MyDictionary = new Dictionary<string, string>();
+
+        public MyClass()
+        {
+            MyDictionary.Remove(""Key"");
+        }
+    }
+}";
+            await VerifyCS.VerifyCodeFixAsync(source, fixedSource);
+        }
+
+        [Fact]
+        public async Task RemoveIsTheOnlyStatementInABlock_ReportsDiagnostic_CS()
+        {
+            string source = @"
+" + CSUsings + @"
+namespace Testopolis
+{
+    public class MyClass
+    {
+        private readonly Dictionary<string, string> MyDictionary = new Dictionary<string, string>();
+
+        public MyClass()
+        {
+            if ([|MyDictionary.ContainsKey(""Key"")|])
+            {
+                MyDictionary.Remove(""Key"");
+            }
```

Comment:
Consider adding a test where there is an `else` and/or `else if`.

e.g:

```csharp
if (MyDictionary.ContainsKey("Key"))
{
    MyDictionary.Remove("Key");
}
else
{
    throw new Exception("Key doesn't exist");
}
```

Reply:
Done. In this PR, it won't report a diagnostic; in @terrajobst's suggestion of later on replacing `ContainsKey` with `Remove` and retaining the condition, it should apply a fixer.